src/documentation/webview/CommunicationBridge.ts

```diff
@@ -0,0 +1,120 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2024 the VS Code Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import { Disposable } from "./Disposable";
+
+/**
+ * Sends and receives messages from swift-docc-render
+ */
+export interface CommunicationBridge {
+    send(message: VueAppMessage): void;
+    onDidReceiveMessage(handler: (message: VueAppMessage) => void): Disposable;
+}
+
+/**
+ * Creates a {@link CommunicationBridge} that can send and receive messages from
+ * swift-docc-render.
+ *
+ * Waits for swift-docc-render to be initialized before resolving.
+ *
+ * @returns A promise that resolves to the created CommunicationBridge
+ */
+export function createCommunicationBridge(): Promise<CommunicationBridge> {
+    if ("webkit" in window) {
+        throw new Error("A CommunicationBridge has already been established");
+    }
+
+    return new Promise<CommunicationBridge>((resolve, reject) => {
+        try {
+            // Define the window.webkit property in order to receive messages
+            const messageHandlers: Set<(message: VueAppMessage) => void> = new Set();
+            Object.defineProperty(window, "webkit", {
```

Comment:
assume the API type is referenced in the swift-docc-renderer project? Maybe good to leave a comment with the link so we know where to start looking if this changes

Reply:
Good point. I added a note to the docstring for createCommunicationBridge to make it clear where the value comes from.